build/flaws.js

```diff
@@ -317,17 +318,27 @@ function injectPreWithHTMLFlaws(level, doc, $, rawContent) {
     // Because Cheerio doesn't support selectors like `pre + code` we have to
     // manually (double) check that the parent really is a `<pre>` tag.
     if ($pre.length && $pre.get(0).tagName === "pre") {
-      addFlaw($pre);
+      addCodeTagFlaw($pre);
     }
   });
 
+  // TODO: Add other <pre> tag flaws underneath.
+  // We chain flaw checks and report only the first flaw kind to esure it's fixable
```

Comment:
I don't know if I understand this. "chain flaws"?
In plain English, the argument is that we only do a maximum 1 flaw check per `<pre>` tag because that flaw might be fixable, and being fixable relies on a string replace which would cease to work for a second fixable flaw.

Reply:
Perhaps replace the `noPreTagFlawsYet()` with:
```javascript
if (!doc.flaws.bad_pre_tags || !doc.flaws.bad_pre_tags.filter(flaw => !flaw.fixable).length) {
  // If there are no fixable flaws so far, proceed with other flaw checks.
  // TODO: Add other <pre> tag flaws underneath.
}
```

I think you can replace the bulk of that if statement with your utility function but now the function needs to mention "fixable". And I know it's a nit, but can you move that function to here instead. Since it's never used for anything else, I'm not so sure it needs to be a function and definitely not that far away from where it's used and matter. 
But it's a nit so feel free to ignore.

Reply:
I think I fixed all the points, how does it look?